src/cli-util.js

```diff
@@ -178,11 +180,21 @@ function applyConfigPrecedence(argv, options) {
   try {
     switch (argv["config-precedence"]) {
       case "cli-override":
-        return parseArgsToOptions(argv, options);
+        return parseArgsToOptions(
+          argv,
+          normalizeConfigs("api", options, constant.detailedOptionMap)
+        );
       case "file-override":
-        return Object.assign({}, parseArgsToOptions(argv), options);
+        return Object.assign(
+          {},
+          parseArgsToOptions(argv),
+          normalizeConfigs("api", options, constant.detailedOptionMap)
+        );
       case "prefer-file":
-        return options || parseArgsToOptions(argv);
+        return (
+          normalizeConfigs("api", options, constant.detailedOptionMap) ||
```

Comment:
Maybe do `normalizeConfigs` (which I would name `normalizeConfig` btw, because with the "s" it sounds like an array of config objects) in the function that calls `applyConfigPrecedence`? It feels a bit odd that `applyConfigPrecedence` also deals with normalizing config, and then we also get rid of the repetition in each `case` of the `switch`.